### PR TITLE
enhance error message if phantomjs executable is not found

### LINKED ISSUE #11868: Make sure Grafana fails nicely when phantomjs is missing
The new crossplatform build will initially only add phantomjs binaries for linux/amd64, darwin and windows. For arm phantomjs will be missing (currently the linux/amd64 version is packaged with the build though).

Grafana needs to fail nicely when phantomjs is missing to support the users running Grafana on arm.

### PR DESCRIPTION
**Pull request by marefr**
Fixes #11868 

if arm build, explain that phantomjs is not included by default in arm builds. If not explain that phantom js isn't installed correctly.

This was the best I could come up with using the existing error page. Please note that the error details section not are displayed if the environment setting of a build = production.

**If arm architecture:**
image

log:
```
INFO[06-07|15:07:39] Rendering                                logger=rendering path="d-solo/RFCYwTSmz/testdata-graph-panel-last-1h?orgId=1&panelId=4&from=1528372675566&to=1528376275566&width
=1000&height=500&tz=Europe%2FStockholm"                                                                                                                                                       
EROR[06-07|15:07:39] executable not found                     logger=rendering executable=/home/marcus/go/src/github.com/grafana/grafana/tools/phantomjs/phantomjs                            
EROR[06-07|15:07:39] Rendering failed - PhantomJS isn't included in arm builds per default logger=context userId=1 orgId=1 uname=admin error="PhantomJS not installed/setup correctly"        
EROR[06-07|15:07:39] Request Completed                        logger=context userId=1 orgId=1 uname=admin method=GET path=/render/d-solo/RFCYwTSmz/testdata-graph-panel-last-1h status=500 remote_addr=[::1] time_ms=4 size=1937 referer=  
```

**If not arm architecture:**
image

log:
```
INFO[06-07|15:10:08] Rendering                                logger=rendering path="d-solo/RFCYwTSmz/testdata-graph-panel-last-1h?orgId=1&panelId=4&from=1528372675566&to=1528376275566&width
=1000&height=500&tz=Europe%2FStockholm"                                                                                                                                                       
EROR[06-07|15:10:08] executable not found                     logger=rendering executable=/home/marcus/go/src/github.com/grafana/grafana/tools/phantomjs/phantomjs                            
EROR[06-07|15:10:08] Rendering failed - PhantomJS isn't installed correctly logger=context userId=1 orgId=1 uname=admin error="PhantomJS executable not found"                                
EROR[06-07|15:10:08] Request Completed                        logger=context userId=1 orgId=1 uname=admin method=GET path=/render/d-solo/RFCYwTSmz/testdata-graph-panel-last-1h status=500 remote_addr=[::1] time_ms=3 size=1913 referer=                                  
```

### CODE REVIEW

File: pkg/api/render.go [hunk 1]
```diff
@@ -55,6 +57,15 @@ func (hs *HTTPServer) RenderToPng(c *m.ReqContext) {
 		return
 	}
 
+	if err != nil && err == rendering.ErrPhantomJSNotInstalled {
+		if strings.HasPrefix(runtime.GOARCH, "armv") {
```
Comment (xlson) [inline pkg/api/render.go]: I think you need to check against `arm`, not `armv`.

See: https://golang.org/doc/install/source#environment

File: pkg/api/render.go [hunk 2]
```diff
@@ -55,6 +57,15 @@ func (hs *HTTPServer) RenderToPng(c *m.ReqContext) {
...
+			c.Handle(500, "Rendering failed - PhantomJS isn't included in arm builds per default", err)
```
Comment (xlson) [inline pkg/api/render.go]: I'd change "included in arm builds" to "included in arm build". Technically we have several arm builds, but for the end user running into this problem only one of them is relevant.

**Review by xlson — CHANGES_REQUESTED** [review]
I was hoping for someone a bit less angry (a yellow warning or something like that). But this is probably fine.

**Comment by torkelo** [comment]
I think it would be ideal if the error message was actually inside an image. The thinking is that the render service would always return an image, so if used in alerting you would get an image that tells you that something went wrong in rendering the image, instead of having to search through logs. But maybe not a good idea, or tricky to do generically (so the error message can be different), maybe 1-3 pre generated error images would also work. 

thoughts?

**Comment by xlson** [comment]
I like that idea. We would probably have to use pre-generated images but that doesn't sound like a bad idea to me.

**Comment by marefr** [comment]
I think I like that idea, but could be tricky. I agree with Torkel that it would be hard to include error message in image why you probably still need to dig into the logs to find the actual problem. 

What would these 1-3 pre generated images be for?
* Rendering not **installed/setup?**
* Rendering timeout?
* Rendering failure?

The easiest thing would be to have only one general pre generated image saying that rendering failed :)

**Comment by torkelo** [comment]
I think 3 images at a minimum. 

* Rendering not installed/setup
* Rendering timeout
* Rendering failed

**Comment by bergquist** [comment]
I think this is good enough for 5.2

We can create a new issue for improving this once this has been released.

**Comment by marefr** [comment]
Decided that this is enough for 5.2

**Pull request merged**